Add evm-bully=yes build to CI

frankbraun opened the pull request:
The contract build for the evm-bully (`make evm-bully=yes`) often breaks.
Add it to CI to prevent that from happening.

sept-en reviewed (approved):
LGTM!

I also think that we should use `develop` as a base branch having the release protocol we have. I.e. the `master` points to the latest deployed version and the `develop` branch is the latest stable one that will be uploaded into `master` and released after that.

artob commented on .github/workflows/builds.yml:
```diff
@@ -61,3 +61,31 @@ jobs:
+          file: bully.wasm
+          overwrite: true
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Update the latest tag
```
You need to remove the `Update the latest release` and `Update the latest tag` steps, lest you introduce a race condition for updating the `latest` tag.

In addition, it is also unlikely that you actually need the `Upload the release.wasm artifact` step (which also wasn't properly renamed).

artob reviewed (changes requested):
This should be targeted at `develop`, not `master`. And there are copy-paste artifacts to remove.

frankbraun commented:
I changed it to target `develop` instead of `master`. Also moved the tests over to `tests.yml`, because this should be checked on every PR, not just for releases.
